core/bin/external_node/src/config/observability.rs

```diff
@@ -1,43 +1,99 @@
-    let sentry_url = if let Ok(sentry_url) = std::env::var("MISC_SENTRY_URL") {
-        if sentry_url == "unset" {
-            None
-        } else {
-            Some(sentry_url)
+use std::{collections::HashMap, time::Duration};
+
+use anyhow::Context as _;
+use prometheus_exporter::PrometheusExporterConfig;
+use serde::Deserialize;
+use vlog::LogFormat;
+
+use super::{ConfigurationSource, Environment};
+
+/// Observability part of the node configuration.
+#[derive(Debug, Default, Deserialize)]
+pub(crate) struct ObservabilityENConfig {
+    /// Port to bind the Prometheus exporter server to. If not specified, the server will not be launched.
+    /// If the push gateway URL is specified, it will prevail.
+    pub prometheus_port: Option<u16>,
+    /// Prometheus push gateway to push metrics to. Overrides `prometheus_port`. A full URL must be specified
+    /// including `job_id` and other path segments; it will be used verbatim as the URL to push data to.
+    pub prometheus_pushgateway_url: Option<String>,
+    /// Interval between pushing metrics to the Prometheus push gateway.
+    #[serde(default = "ObservabilityENConfig::default_prometheus_push_interval_ms")]
+    pub prometheus_push_interval_ms: u64,
+    /// Sentry URL to send panics to.
+    pub sentry_url: Option<String>,
+    /// Environment to use when sending data to Sentry.
+    pub sentry_environment: Option<String>,
+    /// Log format to use: either 'plain' (default) or 'json'.
+    #[serde(default)]
+    pub log_format: LogFormat,
+}
+
+impl ObservabilityENConfig {
+    const fn default_prometheus_push_interval_ms() -> u64 {
+        10_000
+    }
+
+    pub fn from_env() -> envy::Result<Self> {
+        Self::new(&Environment)
+    }
+
+    pub(super) fn new(source: &impl ConfigurationSource) -> envy::Result<Self> {
+        const OBSOLETE_VAR_NAMES: &[(&str, &str)] = &[
+            ("MISC_SENTRY_URL", "EN_SENTRY_URL"),
+            ("MISC_LOG_FORMAT", "EN_LOG_FORMAT"),
+        ];
+
+        let en_vars = source.vars().filter_map(|(name, value)| {
+            let name = name.into_string().ok()?;
+            if !name.starts_with("EN_") {
+                return None;
+            }
+            Some((name, value.into_string().ok()?))
+        });
+        let mut vars: HashMap<_, _> = en_vars.collect();
+
+        for &(old_name, new_name) in OBSOLETE_VAR_NAMES {
+            if vars.contains_key(new_name) {
+                continue; // new name is set; it should prevail over the obsolete one.
+            }
+            if let Some(value) = source.var(old_name) {
+                vars.insert(new_name.to_owned(), value);
+            }
         }
-    } else {
-        None
-    };
-    let sentry_environment = std::env::var("EN_SENTRY_ENVIRONMENT").ok().or_else(|| {
-        let l1_network = std::env::var("CHAIN_ETH_NETWORK").ok();
-        let l2_network = std::env::var("CHAIN_ETH_ZKSYNC_NETWORK").ok();
```

Comment:
AFAICT, these vars are never set for our EN nodes; `EN_SENTRY_ENVIRONMENT` is always used instead; so I've removed this fallback.